ush/python/pygw/src/pygw/fsutils.py

```
@@ -0,0 +1,46 @@
+import os
+import errno
+import shutil
+import contextlib
+
+__all__ = ['mkdir', 'mkdir_p', 'rmdir', 'chdir', 'rm_p']
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise OSError(f"unable to create directory at {path}")
+
+
+mkdir = mkdir_p
+
+
+def rmdir(dir_path):
+    try:
+        shutil.rmtree(dir_path)
+    except OSError as exc:
+        raise OSError(f"unable to remove {dir_path}")
+
+
+@contextlib.contextmanager
+def chdir(path):
+    cwd = os.getcwd()
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(cwd)
```

Is this necessary to recover from the failed `os.chdir(path)`? Otherwise not seeing the point to changing the directory to the one we are in.

Also, I'm not sure about silently failing to change the directory like this. I think a lot of the time we'd want to throw an exception. But maybe I'm misunderstanding how this would be used.

---

we should throw an error if it cannot `chdir` to the directory.

---

On second thought, a `WARNING` for now.  We can raise an exception in the future.